### PR TITLE
vendor: gotest.tools v3.2.0

### PR DESCRIPTION
**Pull request by thaJeztah**
full diff: https://github.com/gotestyourself/gotest.tools/compare/v3.1.0...v3.2.0

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: vendor/gotest.tools/v3/env/env.go [hunk 1]
```diff
@@ -21,6 +21,8 @@ type helperT interface {
 //
 // When used with Go 1.14+ the unpatch function will be called automatically
 // when the test ends, unless the TEST_NOCLEANUP env var is set to true.
+//
+// Deprecated: use t.SetEnv
```
Comment (thaJeztah) [inline vendor/gotest.tools/v3/env/env.go:25]: Taking care of this deprecation in https://github.com/moby/moby/pull/43340

**Review by samuelkarp — APPROVED** [review]

**Comment by thaJeztah** [comment]
https://github.com/moby/moby/pull/43652#discussion_r883759654

> Taking care of this deprecation in #43340

Hmm.. right, so the linter doesn't like this, because of the deprecated function;

```
client/options_test.go:21:8: SA1019: env.Patch is deprecated: use t.SetEnv (staticcheck)
	defer env.Patch(t, "DOCKER_HOST", "tcp://foo.example.com:2376/test/")()
	      ^
client/options_test.go:47:8: SA1019: env.Patch is deprecated: use t.SetEnv (staticcheck)
	defer env.Patch(t, "DOCKER_API_VERSION", "2.9999")()
	      ^
... etc
```

I just separated the `env.Patch()` changes from https://github.com/moby/moby/pull/43340 to a separate commit, and added it to this PR

**Comment by thaJeztah** [comment]
whoop! All green now; @samuelkarp @tianon PTAL

File: client/client_test.go [hunk 2]
```diff
@@ -174,12 +174,12 @@ func TestParseHostURL(t *testing.T) {
...
 
 func TestNewClientWithOpsFromEnvSetsDefaultVersion(t *testing.T) {
-	defer env.PatchAll(t, map[string]string{
+	env.PatchAll(t, map[string]string{
 		"DOCKER_HOST":        "",
 		"DOCKER_API_VERSION": "",
 		"DOCKER_TLS_VERIFY":  "",
 		"DOCKER_CERT_PATH":   "",
-	})()
+	})
```
Comment (samuelkarp) [inline client/client_test.go:182]: Is `env.PatchAll` not deprecated?  Seems like it'd be good to replace this with `t.Setenv` too.

Comment (thaJeztah) [reply client/client_test.go:182]: No, that one's not deprecated, and also does slightly more (which could actually be a nice addition to go's standard tooling;

- it resets the environment (`os.ClearEnv()`)
- patches/sets all of the given env-vars
- restores the environment afterwards

Comment (samuelkarp) [reply client/client_test.go:182]: Was the removal of the `defer` intentional here?  The other calls to `env.PatchAll` in the file had the `defer` and were also changed to `t.Setenv`.

Comment (thaJeztah) [reply client/client_test.go:182]: Yes; newer versions of gotest.tools use `t.CleanUp()` to reset state automatically (I could've left out that part in this PR  when I extracted it from https://github.com/moby/moby/pull/43340)

Comment (samuelkarp) [reply client/client_test.go:182]: Got it, thanks for the explanation!

**Review by samuelkarp — APPROVED** [review]

**Review by rumpl — APPROVED** [review]

**Comment by thaJeztah** [comment]
Thanks all for reviewing; let me bring this one in 👍

**Pull request merged**